src/osd/PG.cc

```diff
@@ -3568,9 +3568,11 @@ void PG::reg_next_scrub()
     return;
 
   utime_t reg_stamp;
+  bool must = false;
   if (scrubber.must_scrub ||
       (info.stats.stats_invalid && cct->_conf->osd_scrub_invalid_stats)) {
     reg_stamp = ceph_clock_now();
+    must = true;
```

Comment:
if `(info.stats.stats_invalid && cct->_conf->osd_scrub_invalid_stats) == 1` and `scrubber.must_scrub == false` still must would become true. Though `scrubber.must_scrub` is false.
So Looks not correct...

Reply:
@amitkumar50 thanks for your review. if (info.stats.stats_invalid && cct->_conf->osd_scrub_invalid_stats) == true, then must become ture, then scrub deadline was set to now in PG::reg_next_scrub(), and the pg will do scrub even though scrubber.must_scrub=false, doesn't this logic anything wrong ?